### ISSUE TITLE
False positives when using experimental rust-analyzer diagnostics

### ISSUE THREAD

**Issue by lalala-233** [rust-lang/rust-analyzer#17782]
**rust-analyzer version**: rust-analyzer version: 0.3.2053-standalone [~/.vscode/extensions/rust-lang.rust-analyzer-0.3.2053-linux-x64/server/rust-analyzer]

**rustc version**: rustc 1.80.0 (051478957 2024-07-21)

**editor or extension**: VSCode

**relevant settings**: `"rust-analyzer.diagnostics.experimental.enable": true,`

**repository link (if public, optional)**: (eg. [rust-analyzer](https://github.com/rust-lang/rust-analyzer))

**code snippet to reproduce**: 

```bash
./
├── Cargo.lock
├── Cargo.toml
└── src
    ├── lib.rs
    └── main.rs
```

```toml
# Cargo.toml
[package]
name = "a"
version = "0.1.0"
edition = "2021"

[features]
default = ["feat"]
feat = []
```

```rust
//lib.rs
pub(crate) use b::*;
#[cfg(feature = "feat")]
pub use b::*;
pub fn test1() {
    println!("1");
}
mod b {
    pub fn test2() {
        println!("2");
    }
}
```

```rust
//main.rs
fn main() {
    use a::*;
    test1();
    test2();
}
```

This is the minimum example I can find.

rust-analyzer will give me an error `no such value in this scope`, but I can run this project.

This is the project's `.tar.gz` file: [a.tar.gz](https://github.com/user-attachments/files/16487066/a.tar.gz)

**Comment by Veykril** [comment]
This works for me, so likely was fixed by https://github.com/rust-lang/rust-analyzer/pull/17715 (will land next monday), if not please re-open then
